Mathlib/Order/Quantale.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2024 Pieter Cuijpers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pieter Cuijpers
+-/
+import Mathlib.Order.CompleteLattice
+import Mathlib.Algebra.Group.Defs
+
+/-!
+# Theory of quantales
+
+Quantales are the non-commutative generalization of locales/frames and as such are linked
+to point-free topology and order theory. Applications are found throughout logic,
+quantum mechanics, and computer science. Intuitively, as described by [vickers1989],
+open sets of a topology form a frame, and can be considered as modeling what is `observable`
+in a system. Quantales come into play when making an observation may change the system itself.
+Traditionally, one would write `x ⊓ y` to describe making observations `x` and `y` at the same
+time, but when the order of observation becomes important due to changes in the system caused
+by the observation, it makes more sense to write `x * y` or `x + y` and consider `*` or `+` as
+a possibly non-commutative monoid.
+
+## Main definitions
+
+* class `Quantale`: a monoid distributing over a complete sup-semilattice.
+  I.e satisfying `x * (sSup s) = ⨆ y ∈ s, x * y` and `(sSup s) * y = ⨆ x ∈ s, x * y`
+
+Conceptually, every complete sup-semilattice is of course a complete lattice.
+However, when considering morphisms between quantales, usually only the supremum
+is required to be preserved. Hence, it is conseptually cleaner to define it as a
+complete sup-semilattice. (In literature, both variants occur.)
```

Comment:
Are you saying that `x * sInf s = ⨅ y ∈ s, x * y` *isn't* always true? Or are you saying that it is, but there are two types of *morphisms* between quantales that are usually considered?